docs(databricks): document the row_filter config for Unity Catalog row-level security (#9374)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/databricks-configs.md

</File>

## Setting row filters
_Available in versions 1.12 or higher_

> Set `row_filter` to apply a [Unity Catalog row filter](https://docs.databricks.com/aws/en/tables/row-and-column-filters) to a model, restricting which rows a query returns based on a SQL UDF. dbt applies the filter with a `WITH ROW FILTER` clause when it creates the relation, and emits `ALTER ... SET ROW FILTER` / `ALTER ... DROP ROW FILTER` to add, update, or remove the filter on subsequent runs.

Reviewer: ```suggestion
You can set `row_filter` to apply a [Unity Catalog row filter](https://docs.databricks.com/aws/en/tables/row-and-column-filters) to a model, restricting which rows a query returns based on a SQL UDF. dbt applies the filter with a `WITH ROW FILTER` clause when it creates the relation, and emits `ALTER ... SET ROW FILTER` / `ALTER ... DROP ROW FILTER` to add, update, or remove the filter on subsequent runs.
```

> `row_filter` is a model-level config with two properties:

Reviewer: ```suggestion
`row_filter` is an optional model-level config. When you set it, both of the following properties are required:
```

| Property   | Description   | Required?| Example  |
|------------|---------------|----------|----------|

> | function   | The row-filter UDF to apply. Provide either an unqualified name (dbt qualifies it with the model's catalog and schema) or a fully qualified `catalog.schema.function`. A two-part `schema.function` name is rejected as ambiguous. | Required | `region_filter` |

Reviewer: ```suggestion
| function   | The row-filter UDF to apply. Provide either an unqualified name (dbt qualifies it with the model's catalog and schema) or a fully qualified `catalog.schema.function`. dbt rejects a two-part schema.function name as ambiguous. | Yes | `region_filter` |
```

> | columns    | The column(s) passed as arguments to the filter function, as a single string or a list. Must be non-empty when `function` is set. | Required | `[region]` |

Reviewer: ```suggestion
| columns    | The columns passed as arguments to the filter function. Can be a single string or a list. Required when `function` is set. | Yes | `[region]` |
```
